### PR TITLE
Draft database for storing goals and descriptions. Bugs in subwindow launching. #12

### PR DESCRIPTION
**Pull request by zztin**
## Types of changes
Please put an `x` in the box that applies

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description
**Describe what the change is**
Record the goals and review into local database.
## Steps to Test This Pull Request
1. type ctimer to command line to initiate ctimer
2. enter a goal for the clock. 
3. Start a clock
4. when a clock ends, click if you finish your goal and enter reasons for why you did or did not finish the goal. (currently the clock is set to very short for testing)
6. Use commandline sqlite to check the recorded entries of the clock.

## Expected behavior
The info is stored in the ctimer.db in the data folder.

## Current Issue NOT Solved
Now the subwindow to allow entering the reasons does not halt the main window. therefore the break could start and finish before the reason is entered. When this happens, the reasons are not recorded in the database. 

## Related issue
Address #12 feature request

### CODE REVIEW

**Comment by tai271828** [comment]
I am able to reproduce this issue with python 3.8.3. At this moment I am looking into the issue.

File: concentratetimer/concentratetimer.py [hunk 1]
```diff
@@ -69,13 +72,47 @@ def create_widgets(self):
 
     def get_goal(self):
         # TODO: get all goals for all clocks for the day
-        self.goal = simpledialog.askstring(title="Set your goals",
-                                           prompt="What's your goal for this clock:")
-        while not self.goal:
```
Comment (tai271828) [inline concentratetimer/concentratetimer.py:74]: My gut feeling shows by removing `while` will result in that ctimer does not wait for dialogue window input.

Comment (tai271828) [reply concentratetimer/concentratetimer.py:74]: hmmm I am probably wrong. Some more complicated logic is used in the pull request. I need more time to follow up your logic.

**Comment by zztin** [comment]
In this draft, the db_file path is fixed to a fixed path. (Previously it was relative path which causes writing errors when launching ctimer at different directory.) The issue with the subwindow launching will be solve in next pull request mbox_fix

**Comment by zztin** [comment]
This is a legacy. Features requests and bugs are addressed in another pull request. Deprecate this pull request.

**Pull request closed without merging**